tests/test_operation.py

```diff
@@ -716,24 +716,45 @@ def test_tensor_matrix(self, tol):
         O = qml.PauliX(0) @ qml.PauliY(1) @ qml.Hermitian(H, [2, 3])
 
         res = O.matrix()
-        expected = np.kron(qml.PauliY.compute_matrix(), H)
-        expected = np.kron(qml.PauliX.compute_matrix(), expected)
+        expected = reduce(np.kron, [qml.PauliX.compute_matrix(), qml.PauliY.compute_matrix(), H])
 
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
     def test_matrix_wire_order_not_implemented(self):
         """Test that an exception is raised if a wire_order is passed to the matrix method"""
         O = qml.PauliX(0) @ qml.PauliY(1)
-        with pytest.raises(NotImplementedError):
+        with pytest.raises(NotImplementedError, match="wire_order"):
             O.matrix(wire_order=[1, 0])
 
-    def test_multiplication_matrix(self, tol):
+    def test_tensor_matrix_partial_wires_overlap_warning(self, tol):
+        """Test that the tensor product matrix method returns
+        the correct result if the observables are added unsorted
+        with respect to a canonical wire ordering."""
+        H = np.diag([1, 2, 3, 4])
+        O1 = qml.PauliX(0) @ qml.Hermitian(H, [0, 1])
+        O2 = qml.Hermitian(H, [0, 1]) @ qml.PauliY(1)
+
+        for O in (O1, O2):
+            with pytest.warns(UserWarning, match="partially overlapping"):
+                O.matrix
+
+    def test_tensor_matrix_too_large_warning(self, tol):
+        """Test that the tensor product matrix method returns
+        the correct result if the observables are added unsorted
+        with respect to a canonical wire ordering."""
```

Comment:
sounds like this docstring should be updated?

Reply:
(as it doesn't seem to match the test)

Reply:
Oh, yes! Thanks :)